src/VisualStudio/Core/Def/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -163,6 +164,18 @@ private abstract class AbstractTableDataSourceFindUsagesContext :
                 this.UpdateTableProgressAsync,
                 presenter._asyncListener,
                 CancellationTokenSource.Token);
+
+            // Similarly, updating the actual FAR window can be quite expensive (especially when there are thousands of
+            // results).  To limit the amount of work we do, we'll only update the window every 500ms.
+            _notifyQueue = new AsyncBatchingWorkQueue(
+                DelayTimeSpan.Medium,
```

Comment:
>DelayTimeSpan.Medium [](http://example.com/codeflow?start=16&length=20)

I kind of feel like I wish the ABWQ had an option where it would customize the delay as the work expanded.

Something like:
For the first 1 second, it updates on NearImmediate cadence.
Between 1 - 5 seconds, it updates on short cadence
After 5 seconds, it updates on medium cadence

It just feels like adding a 500 ms delay on really quick FAR requests might be perceptible.

Reply:
I def didn't notice anything feeling off here. I could make it 250ms if you feel better about that. We'll still likely get the majority of the benefit.

Reply:
I would like that. 250 ms is a lot less noticeable than 500 ms.